mobile/android/app/src/main/kotlin/app/alextran/immich/MainActivity.kt

```diff
@@ -1,15 +1,68 @@
 package app.alextran.immich
 
+import android.content.ContentValues
+import android.os.Build
+import android.provider.MediaStore
+import android.util.Log
+import androidx.annotation.NonNull
 import io.flutter.embedding.android.FlutterActivity
 import io.flutter.embedding.engine.FlutterEngine
-import android.os.Bundle
-import android.content.Intent
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import java.io.File
 
 class MainActivity : FlutterActivity() {
+    private val CHANNEL = "file_trash"
 
-    override fun configureFlutterEngine(flutterEngine: FlutterEngine) {
+    override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
         super.configureFlutterEngine(flutterEngine)
         flutterEngine.plugins.add(BackgroundServicePlugin())
+
+        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, CHANNEL).setMethodCallHandler { call, result ->
+            if (call.method == "moveToTrash") {
+                val filePath = call.argument<String>("filePath")
+                if (filePath != null) {
+                    val success = moveToTrash(filePath)
+                    result.success(success)
+                } else {
+                    result.error("INVALID_PATH", "The file path is not specified.", null)
+                }
+            } else {
+                result.notImplemented()
+            }
+        }
     }
 
+    private fun moveToTrash(filePath: String): Boolean {
+    return try {

```

Comment:
I'm not that familiar with this mobile codebase, but it kind of looks like this line and everything below it need to be indented an additional level